app/frontend/src/utils/hasAtLeastOnePage.ts

```diff
@@ -0,0 +1,20 @@
+import { InfiniteData } from "react-query";
+
+type EndsWithList = `${string}List`;
+type ObjectWithListValue<TData> = {
+  [Key in keyof TData]: Key extends EndsWithList ? unknown[] : unknown;
+};
+
```

Comment:
This is quite possibly the most bit of TypeScript black magic I managed to fit in around 15 lines of code (minus the comments)! The cool bit about this is since our list data always end with `List` from the generated types, this takes advantage of that constraint on the type for `listKey`. The result is we get autocomplete values in VS Code telling you exactly which key(s) you can use 🤯